core/src/scene/importer.cpp

```diff
@@ -56,10 +57,8 @@ Node Importer::applySceneImports(Platform& platform) {
                 continue;
             }
 
-            if (m_importedScenes.find(nextUrlToImport) != m_importedScenes.end()) {
-                // This scene URL has already been imported, we're done!
-                continue;
-            }
+            // Mark Url as going-to-be-imported to prevent duplicate work.
+            m_importedScenes.emplace(nextUrlToImport, std::make_pair<Node, std::vector<Url>>(Node(), {}));
```

Comment:
nit:
`The element may be constructed even if there already is an element with the key in the container, in which case the newly constructed element will be destroyed immediately.` - https://en.cppreference.com/w/cpp/container/unordered_map/emplace

Might still make sense to check for the existence of the key before calling emplace to avoid this construction/destruction!

Reply:
The same url cannot be added twice to `m_sceneQueue`, see `addSceneYaml` - so I would think this is safe